### DISCUSSION TITLE
下载

### DISCUSSION THREAD

**Discussion by CHG5588** [Meloong-Git/PCL#4495] [提问]
用pcl2下载java21时会失败。

**Comment by allMagicNB** [comment]
# 提问价位表
| 描述类型 | 金额 |
| - | :-: |
| 20 字以内描述 | 500 元 |
| 详细描述+截图+日志+有礼貌 | 免费解决 |

**Comment by WForst-Breeze** [comment]
可以把问题说的详细一点点嘛，没太听懂 😭

**Comment by CHG5588** [comment]
就是用pcl2下载JAVA21时突然返回主界面。什么也没有提示。由于我电脑没有JAVA21，pcl2就在启动mc过程中，顺便帮我下载JAVA21。但是在下载到一半的时候突然返回主界面，什么提示都没有。点启动游戏I又让我提示下载JAVA21。点了好几次都这样的。

**Reply by WForst-Breeze** [reply]
可否上传一下日志呢？

**Reply by MoYuan-CN** [reply]
上传你的 PCL 日志，它应当位于 Plain Craft Launcher 2.exe 文件同目录下 PCL 文件夹内。
如果你知道是哪个日志，上传他。如果你不知道，请将 Log1~5.txt 都上传上来

**Comment by CHG5588** [comment]
[Log3.txt](https://github.com/user-attachments/files/16612793/Log3.txt)
[Log4.txt](https://github.com/user-attachments/files/16612794/Log4.txt)
[Log5.txt](https://github.com/user-attachments/files/16612795/Log5.txt)
[Log1.txt](https://github.com/user-attachments/files/16612796/Log1.txt)
[Log2.txt](https://github.com/user-attachments/files/16612797/Log2.txt)

**Reply by Chiloven945** [reply]
~提供的日志似乎没有关于下载Java的信息？~ 不过判断应该是网络环境不佳的问题，请尝试使用VPN、代理或其他能够改善网络环境的工具，或检查你的网络。
```log
[20:02:32.767] 加载线程 获取 Java 下载信息 (571) 发生运行时错误：连接服务器超时（第一下载源：https://bmclapi2.bangbang93.com/v1/products/java-runtime/2ec0cc96c44e5a76b9c8b7c39df7210883d12871/all.json）
你的网络环境不佳，导致难以连接到服务器。请检查网络，多重试几次，或尝试使用 VPN。
————————————
详细错误信息：
   在 PCL.ModLoader.LoaderBase.WaitForExitTime(Int32 Timeout, Object Input, String TimeoutMessage, Object LoaderToSyncProgress, Boolean IsForceRestart)
   在 PCL.ModNet.NetGetCodeByDownload(IEnumerable`1 Urls, Int32 Timeout, Boolean IsJson, Boolean UseBrowserUserAgent)
   在 PCL.ModJava.JavaFileList(LoaderTask`2 Loader)
   在 PCL.ModLoader.LoaderTask`2._Closure$__13-0._Lambda$__0()
错误类型：System.TimeoutException
```

**Comment by wuliaodexiaoluo** [comment]
>就是用pcl2下载JAVA21时突然返回主界面

根据 Author 的描述
如果是自动补全的话，这个下载行为应该是完全静默的
~因为不会自动跳转到下载页面~
开着下载界面应该是有提示的，至少不可能出现突然返回主界面这种情况，最起码会有个弹窗
```
[20:02:32.767] 加载线程 获取 Java 下载信息 (571) 发生运行时错误：连接服务器超时（第一下载源：https://bmclapi2.bangbang93.com/v1/products/java-runtime/2ec0cc96c44e5a76b9c8b7c39df7210883d12871/all.json）
你的网络环境不佳，导致难以连接到服务器。请检查网络，多重试几次，或尝试使用 VPN。
————————————
详细错误信息：
   在 PCL.ModLoader.LoaderBase.WaitForExitTime(Int32 Timeout, Object Input, String TimeoutMessage, Object LoaderToSyncProgress, Boolean IsForceRestart)
   在 PCL.ModNet.NetGetCodeByDownload(IEnumerable`1 Urls, Int32 Timeout, Boolean IsJson, Boolean UseBrowserUserAgent)
   在 PCL.ModJava.JavaFileList(LoaderTask`2 Loader)
   在 PCL.ModLoader.LoaderTask`2._Closure$__13-0._Lambda$__0()
错误类型：System.TimeoutException
```
~这难道就是传说中的电话线拨号上网？？？~
